### PR TITLE
Vs eng harpoon can shoot through every wall of the ship lv 478

### PR DESCRIPTION
**Pull request by DavidHenvick**
jira: https://bradleycapstone.atlassian.net/browse/LV-486?atlOrigin=eyJpIjoiOWExMDdkZGY1ZDJkNDc2MGIzODMyOGNjZDVjNDVmYjMiLCJwIjoiaiJ9

how to test: 
go into main scene, fire, see that it doesn't go through walls anymore : )

### CODE REVIEW

File: Assets/GameAssets/Player/PlayerScripts/Harpoon/HarpoonProjectileMovement.cs [hunk 1]
```diff
@@ -66,4 +67,16 @@ private void HarpoonFiredProjectileMovement(Vector3 movement)
...
         transform.position += movement;
     }
+
+    /// <summary>
+    /// Used to destroy the harpoon projectile when it hits something in scene.
+    /// </summary>
+    /// <param name="block"></param> what it collides against. 
+    private void OnTriggerEnter(Collider block)
+    {
+        if(block.gameObject.name != "WeakPoint(Clone)")
```
Comment (AlecatEnjoyer) [inline Assets/GameAssets/Player/PlayerScripts/Harpoon/HarpoonProjectileMovement.cs:77]: use block.gameobject.tag != "Whatever weakpoint's tag is" instead so yo don't have to worry too much about clones

**Review by AlecatEnjoyer — APPROVED** [review]
I approved but please do comment if possible. I originally made the weakpoints to have tags on it but if not you can assign one to it.

File: Assets/GameAssets/Player/PlayerScripts/Harpoon/HarpoonProjectileMovement.cs [hunk 2]
```diff
@@ -66,4 +67,16 @@ private void HarpoonFiredProjectileMovement(Vector3 movement)
...
         transform.position += movement;
     }
+
+    /// <summary>
+    /// Used to destroy the harpoon projectile when it hits something in scene.
+    /// </summary>
+    /// <param name="block"></param> what it collides against. 
+    private void OnTriggerEnter(Collider block)
+    {
+        if(block.gameObject.tag != "enemy")
```
Comment (ASLH68) [inline Assets/GameAssets/Player/PlayerScripts/Harpoon/HarpoonProjectileMovement.cs:77]: `if(!block.TryGetComponent<WeakPoint>(out Weakpoint weakpoint))` instead is much less likely to break or cause errors than a tag

**Review by ASLH68 — CHANGES_REQUESTED** [review]

**Review by astapay — APPROVED** [review]
Yippee!

**Pull request merged**